### DISCUSSION TITLE
Is there any way to load all icons?

### DISCUSSION THREAD

**Discussion by Bhautik0110** [unocss/unocss#3317] [Q&A]
Hello there,

**Expected behaviour:**
Currently, We are loading icons based on API response return. Our final bundle does not include dynamic icons. Is there any way to load all icons and include them so that all dynamic icons work?

Thanks.

**Comment by zyyv** [comment]
You only need to download the utility of all the icons you need and put them into the safelist. Then, those dynamic icons will be work.

**Comment by Bhautik0110** [comment]
I want to do without a safelist. Wildcard. Is it possible? thanks.

**Comment by zyyv** [comment]
> Our final bundle does not include dynamic icons

As you said, if the dynamic icons not in your bundle, and because of uno extract utilities from static template, uno how to extractor them?

**Comment by Bhautik0110** [comment]
What I want to say
> Here when I am rendering the icon from API response, it shows empty in dev as well.

**Comment by chu121su12** [comment]
Recent version exports the icon list you can then download
https://github.com/unocss/unocss/blob/b18601bf3e510e57242b98984c6ac04402fa3d9a/packages/preset-icons/src/core.ts#L16
```js
const url = `https://raw.githubusercontent.com/iconify/icon-sets/master/json/${icon}.json`;
await fetch(url)
    .then(x => x.text())
    .then(x => fs.writeFileSync(`${ASSET_DIR}/icons/${icon}.json`, x));
```

At the same time, use customFetch to alter the download:
```js
presetIcons({
    customFetch: (url) => {
        // local url
        const iconUrl = `${BASE_URL}/assets/icons/${url.replace(/.*\/([^/]+)\/.*/, '$1')}.json`;
        // or maybe add local cache here
        return fetch(iconUrl).then(data => data.json());
    },
}),
```
